crates/ra_progress/src/lib.rs

```diff
@@ -0,0 +1,129 @@
+//! General-purpose instrumentation for progress reporting.
+//!
+//! Note:
+//! Most of the methods accept `&mut self` just to be more restrictive (for forward compat)
+//! even tho for some of them we can weaken this requirement to shared reference (`&self`).
```

Comment:
Not sure if this is the right thing. Logically the methods have to be `&mut self`, but since `Sender` uses internal mutability under the hood, this is not required.